apps/api/src/domain/models/bibliographic-reference/entities/book-bibliographic-reference.entity.ts

```diff
@@ -0,0 +1,23 @@
+import { DTO } from '../../../../types/DTO';
+import { resourceTypes } from '../../../types/resourceTypes';
+import { isNullOrUndefined } from '../../../utilities/validation/is-null-or-undefined';
+import { Resource } from '../../resource.entity';
+import { IBibliographicReference } from '../interfaces/IBibliographicReference';
+import BookBibliographicReferenceData from './BookBibliographicReferenceData';
+
+export class BookBibliographicReference
+    extends Resource
+    implements IBibliographicReference<BookBibliographicReferenceData>
+{
+    readonly type = resourceTypes.bibliographicReference;
```

Comment:
What I'm not understanding here is why we would need a type here within the `BookBibliographicReference` that says it's a `bibliographicReference` resource type?  Wouldn't that be implicit?

Reply:
You **never** want to do implicit type discrimination. A DTO for any resource should always be able to be explicitly matched to its `ResourceType`  (/collection) via the type property. 

We call polymorphic methods such as `getCompositeIdentifier`  (see `resource.entity.ts`) that simply will not work without it.

Now maybe you are thinking that the `type` is somehow inherited. It's important to note that there is no base class for BibliographicReference- only an interface. This prevents us from getting caught up in paying the `inheritance tax` game.